feat:MainPage

ouo314 opened the pull request:
## 變更內容
- 新增 MainPage 並分成以下元件（AboutSection、ClassSection、HeroSection、JoinSection、ShowcaseSection）
- 新增 assets 目錄下所需 GIF、icon 圖檔(暫時展示功能用)
- 響應式設計使用vw處理
##其他
- 如果檔案結構或coding style有問題還請不吝指教

ouo314 committed:
feat:Mainpage

ouo314 committed:
feat:Mainpage

dee9c-git commented on src/assets/MainpageHeroBackground1.jpg:
我自己會把Mainpage -> MainPage
e.g. MainPageHeroBackground1

dee9c-git commented on src/components/MainPage/AboutSection.tsx:
```diff
@@ -0,0 +1,34 @@
+                    Tricking = Taekwondo + Gymnastics Flips + Breaking
+                </p>
+                <p className={styles.about_content}>
+                    More accurately,<br />
+                    Tricking is an extreme sport that combines elements of gymnastics,<br />
+                    martial arts, and breaking. Its main elements include flips, kicks,<br />
+                    twists, and ground movements. It is presented in a fluid and visually<br />
+                    impactful manner.
```
若手動以`<br />`換行可能會讓螢幕大小不同的電腦手機觀感不佳，可以給about_content一個width看看可不可以達到同樣效果

ouo314 replied:
第一行太短了 不換行第二行會跑上去 我用F12看應該是還好

dee9c-git commented on src/components/MainPage/HeroSection.tsx:
```diff
@@ -0,0 +1,52 @@
+import img1 from '../../assets/MainpageHeroBackground1.jpg';
+import img2 from '../../assets/MainpageHeroBackground2.jpg';
+
+const images = [img1, img2, img1, img2];
```
目前看起來`const images = [img1, img2];`即可

ouo314 replied:
我看figma是4個就先用4個了

dee9c-git commented on src/pages/MainPage.tsx:
```diff
@@ -1,10 +1,20 @@
 function MainPage() {
     return <>
         <Header />
         <p> Hi there.</p>
```
我當時亂寫的，不用惹XD

dee9c-git commented on src/components/MainPage/AboutSection.tsx:
```diff
@@ -0,0 +1,34 @@
+
+
+            <iframe className={styles.about_video}
+                src="https://www.youtube.com/embed/cevKK6uuCfU"
```
這樣好像是可以的，styles.about_video能讓人知道這是說明影片，但還是有點像magic link。
可以`tricking_explaination_src="https: ..."`後`src=tricking_explaination_src`試試

New9c reviewed (approved):
棒棒👍
喔
`##`和`其他`中間記得空格
可以利用Preview看md結果的樣子

zhihao1021 commented on src/components/MainPage/ClassSection.tsx:
```diff
@@ -0,0 +1,38 @@
+const ClassSection = () => {
+    return (
+        <div className={styles.class_section}>
+            <div styles="text-align:center">
```
這個 Style 要換成 SCSS Module

ouo314 committed:
fix:Mainpage
-AboutSection: magic link
-ClassSection: inline style ->scss
-change file name Main"P"age
-fix button CSS
